SourceX/DiabloUI/credits.cpp

```diff
@@ -33,31 +33,26 @@ struct SurfaceDeleter {
 	}
 };
 
-using SurfacePtr = std::unique_ptr<SDL_Surface, SurfaceDeleter>;
-
 struct CachedLine {
-	CachedLine() = default;
-
-	explicit CachedLine(std::size_t index, SurfacePtr surface)
-	    : index(index)
-	    , surface(std::move(surface))
-	    , palette_version(pal_surface_palette_version)
+	CachedLine(std::size_t index, SDL_Surface *surface)
 	{
+		m_index = index;
+		m_surface = surface;
 	}
 
-	std::size_t index;
-	SurfacePtr surface;
+	std::size_t m_index;
+	SDL_Surface *m_surface;

```

Comment:
`SurfacePtr` handled surface deletion. With a regular pointer, deletion now has to be done manually.

Can add accessor methods and put `delete` there and the destructor.

Reply:
I'm not quite sure how to do that, at least not at this time.

I have applied your other suggestions as best able.

Reply:
Got some help from @Lundar in solving this :)

Tests look good so I have now merged this.